job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/S3EntryContext.groovy

```diff
@@ -31,4 +33,12 @@ class S3EntryContext implements Context {
     void managedArtifacts(boolean managedArtifacts = true) {
         this.managedArtifacts = managedArtifacts
     }
+
+    void useServerSideEncryption(boolean useServerSideEncryption = true) {
```

Comment:
add `@since 1.38` JavaDoc tags and a `@RequiresPlugin` annotation with a minimum version for each method. Either use the latest plugin version of the S3 plugin as minimum vesion or figure out the version which added the features.